x-pack/plugins/security_solution/server/lib/detection_engine/signals/executors/threshold.ts

```diff
@@ -122,6 +125,18 @@ export const thresholdExecutor = async ({
       lists: exceptionItems,
     });
 
+    // If we have a timestampOverride, we'll compute a runtime field that emits the override for each document if it exists,
+    // otherwise it emits @timestamp. If we don't have a timestamp override we don't want to pay the cost of using a
+    // runtime field, so we just use @timestamp directly.
+    const { timestampField, timestampRuntimeMappings } = ruleParams.timestampOverride
```

Comment:
Now that we have the "disable timestamp fallback" flag we may want to condition this logic on `secondaryTimestamp` existing instead of `ruleParams.timestampOverride` directly - this way if a timestamp override is defined but fallback is disabled we can still avoid using the runtime field.

It may also make sense to pull this logic out into the security rule type wrapper and provide it as an "aggregatable timestamp field" to the specific rule types so new terms and threshold can share the logic more easily.